src/test/java/cat/udl/eps/softarch/mytournamentx/steps/JoinToTeamStepDefs.java

```diff
@@ -0,0 +1,142 @@
+import cucumber.api.java.en.And;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.RestMediaTypes;
+import org.springframework.http.MediaType;
+
+public class JoinToTeamStepDefs {
+
+    @Autowired
+    private TeamRepository teamRepository;
+
+    @Autowired
+    private PlayerRepository playerRepository;
+
+
+    private Team team;
+    private Team team2;
+
+    @Autowired
+    private StepDefs stepDefs;
+
+    private TeamInvitation teamInv;
+
+    @When("^I try to join a not created team with name \"([^\"]*)\"$")
+    public void iTryToJoinTeamWithName(String name) throws Throwable {
+        stepDefs.result = stepDefs.mockMvc.perform(
+                get("/teams/{name}",name)
+                .accept(MediaType.APPLICATION_JSON_UTF8))
+                .andExpect(status().isNotFound());
+    }
+
+    @And("^I cannot join the team with name \"([^\"]*)\"$")
+    public void iCannotJoinTheTeamWithName(String name) throws Throwable {
+        stepDefs.result = stepDefs.mockMvc.perform(
+                get("/teams/{name}",name)
+                        .accept(MediaType.APPLICATION_JSON_UTF8))
+                .andDo(print())
+                .andExpect(status().isNotFound());
+    }
+
+    @And("^I can join the team with name \"([^\"]*)\"$")
+    public void iCanJoinTheTeamWithName(String name) throws Throwable {
+        team = teamRepository.findTeamByName(name);
+        stepDefs.result = stepDefs.mockMvc.perform(
+                post("teams/{name}", name)
+                        .contentType(MediaType.APPLICATION_JSON_UTF8)
+                        .accept(MediaType.APPLICATION_JSON_UTF8)
+                        .with(authenticate()))
+                .andDo(print());
+    }
+
+
+
+    @And("^I already joined a team with name \"([^\"]*)\", game \"([^\"]*)\", level \"([^\"]*)\", maxPlayers (\\d+), and the team leader is \"([^\"]*)\"$")
+    public void iAlreadyJoinedATeamWithNameGameLevelMaxPlayersAndTheTeamLeaderIs(String name, String game, String level, int maxPlayers, String demoP) throws Throwable {
+        team = new Team();
+        String message = stepDefs.mapper.writeValueAsString(team);
+        stepDefs.result = stepDefs.mockMvc.perform(
+                post("/teams/{name}",name)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(message)
+                        .accept(MediaType.APPLICATION_JSON)
+                        .with(authenticate()))
+                .andDo(print());
+    }
+
+    @When("^I leave the team with name \"([^\"]*)\", game \"([^\"]*)\", level \"([^\"]*)\", maxPlayers (\\d+), and the team leader is \"([^\"]*)\"$")
+    public void iLeaveTheTeamWithNameGameLevelMaxPlayersAndTheTeamLeaderIs(String name, String game, String level, int maxPlayers, String demoP) throws Throwable {
```

Comment:
You are removing the whole team instead of removing a particular player (yourself) from it. You should do a DELETE to /teams/{teamId}/players/{playerId}